docs/doxygen/migration/mqtt.txt

```diff
@@ -0,0 +1,127 @@
+/**
+@page mqtt_migration MQTT
+@brief How to migrate an MQTT application from v3 to 202009.00.
+
+The MQTT library has been redesigned in 202009.00 as a solely single-threaded library. New features such as persistent session support and QoS 2 publish delivery have also been added.
+
+The features of the MQTT libraries in v3 and 202009.00 are defined by the common MQTT 3.1.1 spec; therefore, the two versions are similar.
+
+@section migration_mqtt_removed Removed features
+@brief The following features were present in v3, but removed in 202009.00:
+- <u>Auto-reconnect</u> <br>
+  Version 3 has an auto-reconnect feature triggered through a call to `aws_iot_mqtt_yield`. In version 202009.00, `aws_iot_mqtt_yield` has been removed, as well as the auto-reconnect feature. <br>
+  <b>Workaround:</b> When a version 202009.00 API call does not return @ref MQTTSuccess, the application should re-establish
+  the underlying transport connection and then call @ref MQTT_Connect. The application should employ an exponential backoff strategy when re-establishing connections.
+- <u>Thread support</u> <br>
+  Version 3 has an `_ENABLE_THREAD_SUPPORT_` macro for multi-threading use cases. Version 202009.00 has been redesigned to use only a single execution thread, and is
+  not thread-safe.
```

Comment:
Why is this on a separate line?

Reply:
Calling it not thread-safe sounds misleading. How about?

Version 202009.00 does not spawn any thread and therefore, the macro `_ENABLE_THREAD_SUPPORT_` has been removed.

Reply:
Changed. Does the newline matter? It doesn't affect formatting in any way.